### PR TITLE
Fix fatalErrors by adding Swift errors

### PR DESCRIPTION
**Pull request by jhugman**
This PR fixes `fatalError`s which are routinely encountered in unrecognized routes.

It has a breaking API change, which may not be acceptable to existing clients.

The tests and example project has been updated to reflect these changes.

### CODE REVIEW

File: Source/SwiftRouter.swift [hunk 1]
```diff
@@ -87,6 +90,8 @@ extension String {
     }
 }
 
+
+
```
Comment (justindarc) [inline Source/SwiftRouter.swift]: nit: extra newline

File: Source/SwiftRouter.swift [hunk 2]
```diff
@@ -187,22 +189,22 @@ open class Router {
                     let key = s.substring(from: s.index(s.startIndex, offsetBy: 1))
                     params[key] = pathComponent
                     if pathComponent == pathComponents.last {
-                        return (v as? NSDictionary)?[kRouteEntryKey] as? RouteEntry
+                        return (v as? NSDictionary)?[kRouteEntryKey] as! RouteEntry
```
Comment (justindarc) [inline Source/SwiftRouter.swift:193]: Does this need to be a forced cast?

File: SwiftRouterTests/SwiftRouterTests.swift [hunk 3]
```diff
@@ -18,6 +18,8 @@ class SwiftRouterTests: XCTestCase {
         super.tearDown()
     }
 
+
+
```
Comment (justindarc) [inline SwiftRouterTests/SwiftRouterTests.swift]: nit: extra newline

**Comment by skyline75489** [comment]
Thanks for the PR. The CI build seems to be failing because it's using Xcode 7.2. Would you mind correct that?

**Comment by skyline75489** [comment]
Looks good. I'll have a look at the tests later.

**Pull request merged**